actions/aws-upload.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * The method exported from this file is used to analyze options and environment variables
+ * to tar the contents of a directory and upload the tarball to an AWS S3 file system.
+ *
+ * This is an example to use this method straight from the command line (fish shell):
+ *
+ * env
+ *  TEST=true
+ *  AWS_ACCESS_KEY_ID=<insert your AWS access key here>
+ *  AWS_SECRET_ACCESS_KEY=<insert your AWS secret access key here>
+ *  AWS_PUBLIC=<set this to true to make the uploaded file public>
+ *  AWS_BUCKET=<set this to bucketName/path/to/file/filename.tar.gz
+ *  AWS_UPLOAD=<set this to the directory you want uploaded>
```

Comment:
Is this a source directory? If so, then perhaps we could rename it to be a little clearer.